yaml.go

```diff
@@ -96,6 +96,62 @@ func JSONToYAML(j []byte) ([]byte, error) {
 	return yaml.Marshal(jsonObj)
 }
 
+// internalYamlUnmarshal works like yaml.Unmarshal but supports streams.
+func internalYamlUnmarshal(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(false, b, out)
+}
+
+// internalYamlUnmarshalStrict works like yaml.UnmarshalStrict but supports streams.
+func internalYamlUnmarshalStrict(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(true, b, out)
+}
+
+// internalYamlUnmarshalWrapper is a wrapper around yaml.Decoder that allows
+// for decoding yaml streams.
+func internalYamlUnmarshalWrapper(strict bool, b []byte, out interface{}) error {
+	if len(bytes.TrimSpace(b)) == 0 {
+		out = nil
```

Comment:
I realized this has no effect and I still need to use reflect here and down on 144 too. Sending a new commit for that.